public/coda/firestore.rules

```diff
@@ -0,0 +1,16 @@
+service cloud.firestore {
+  match /databases/{database}/documents {
+    match /{document=**} {
+        function userIsAllowed(datasetResource){
+           return request.auth.token.email in datasetResource.data.users 
+        }
+        function getDatasetResource(){
+            return get(/databases/$(database)/documents/$(documents))
+        }
+        allow read: if userIsAllowed(resource)
+      match /metrics/{metrics=**} {
+      	allow read, write: if userIsAllowed(getDatasetResource())
+      }
+    }
+  }
+  
```

Comment:
Sorry @IsaackMwenda I don't think this works, I'm going to remove it from the PR